src/sentry/replays/lib/query.py

```diff
@@ -92,6 +92,33 @@ def as_condition(
         return Condition(Column(self.query_alias or self.attribute_name), operator, value)
 
 
+class UUIDField(Field):
+    """
+    as our minimum supported clickhouse version is 20.3, we don't have
+    isUUIDOrZero function, so we must validate the uuid before supplying to clickhouse.
+    """
+
+    _operators = [Op.EQ, Op.NEQ, Op.IN, Op.NOT_IN]
+    _python_type = str
+
+    def as_condition(
+        self, field_alias: str, operator: Op, value: Union[List[str], str], is_wildcard: bool
+    ) -> Condition:
+        if isinstance(value, list):
+            uuids = _transform_uuids(value)
+            if uuids is None:
+                return Condition(Function("identity", parameters=[1]), Op.EQ, 2)
+            value = [Function("toUUID", parameters=[v]) for v in value]
+        else:
+            uuid = _transform_uuids([value])
+            if uuid is None:
+                return Condition(Function("identity", parameters=[1]), Op.EQ, 2)
+
+            value = Function("toUUID", parameters=uuid)
```

Comment:
```suggestion
            value = Function("toUUID", parameters=uuid[0])
```

Needs to use the value in the first index.  Not the best API I know.

Reply:
I think the current code is correct: value will only be one element, so `transform_uuids` will return a list w/ one element. `parameters` expects a list, not a scalar value. your suggestion results in:


```
sentry.utils.snuba.QueryExecutionError: DB::Exception: Incorrect number of arguments for function toUUID provided 36 (String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String, String), expected 1 ('Value' : ): While processing SELECT project_id AS _snuba_project_id, replay_id AS _snuba_replay_id, notEmpty(groupArray(is_archived AS _snuba_is_archived)) AS _snuba_isArchived, groupArray(1)(environment AS _snuba_environment)[1] AS _snuba_agg_environment, min(replay_start_timestamp AS _snuba_replay_start_timestamp) AS _snuba_started_at, _snuba_replay_id, max(timestamp AS _snuba_timestamp) AS _snuba_finished_at FROM replays_local WHERE ((_snuba_project_id IN [4551592305098753]) AND (_snuba_timestamp < toDateTime('2023-03-15T19:29:07', 'Universal')) AND (_snuba_timestamp >= toDateTime('2022-12-15T19:29:07', 'Universal'))) AND ((_snuba_replay_id = toUUID('f', '3', '9', '0', '6', '8', 'f', '6', '-', '0', 'f', 'b', 'e', '-', '4', '5', 'a', '0', '-', 'a', '8', '9', '9', '-', 'a', '5', '3', '6', 'e', '3', 'a', '2', '2', 'a', '6', 'f')) OR (_snuba_replay_id = toUUID('2', '8', '1', '7', '3', 'b', '1', '9', '-', '4', '1', '1', '8', '-', '4', '9', '6', '1', '-', '9', '5', '8', '0', '-', '9', 'a', '6', '5', '8', '3', '8', '4', 'a', '8', '1', '7')) OR (_snuba_replay_id = toUUID('e', '8', 'd', 'a', 'e', '8', '4', 'a', '-', '9', 'c', 'a', '9', '-', '4', '6', '0', '8', '-', '8', '1', '6', 'c', '-', '7', '5', '7', '7', '9', 'd', '2', '4', '9', 'a', 'd', '6'))) GROUP BY _snuba_project_id, _snuba_replay_id HAVING ((min(segment_id AS _snuba_segment_id) = 0) AND (_snuba_finished_at < toDateTime('2023-03-15T19:29:07', 'Universal'))) AND (_snuba_isArchived = 0) ORDER BY _snuba_started_at DESC LIMIT 0, 10. Stack trace:
````